libs/contact/src/effects/contact.effects.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable, Inject } from '@angular/core';
+import { DaffContactDriver } from '../driver/injection-tokens/contact-driver.token';
+import { DaffContactServiceInterface } from '../driver/interfaces/contact-service.interface';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { Action } from '@ngrx/store';
+import { Observable, of } from 'rxjs';
+import { DaffContactActionTypes, DaffContactSubmit, DaffContactCancel, DaffContactSuccessSubmit, DaffContactFailedSubmit, DaffContactRetry } from '../actions/contact.actions';
+import { switchMap, map, catchError } from 'rxjs/operators';
+
+@Injectable()
+export class DaffContactEffects<T, V>{
+  constructor(
+    private actions$: Actions,
+    @Inject(DaffContactDriver) private driver: DaffContactServiceInterface<T, V>
+  ) { }
+
+  trySubmission$: Observable<Action> = createEffect(() => this.actions$.pipe(
+    ofType(DaffContactActionTypes.ContactSubmitAction,
+      DaffContactActionTypes.ContactRetryAction,
+      DaffContactActionTypes.ContactCancelAction),
+    switchMap((action: DaffContactSubmit<T> | DaffContactRetry<T> | DaffContactCancel) => {
+      if ((action.type === DaffContactActionTypes.ContactCancelAction)) {
+        return of(action);
+      }
+      else if (action instanceof DaffContactSubmit || action instanceof DaffContactRetry) {
+        return this.driver.send(action.payload).pipe(
+          map((resp: V) => {
+            return new DaffContactSuccessSubmit();
+          }),
+          catchError(error => {
+            return of(new DaffContactFailedSubmit(['Failed to submit']));
+          })
+        )
```

Comment:
Could you move the this.driver.send call into a private function to simplify the effect a bit? Like
```
private submitContact(action: Action): Action {
   return this.driver.send(action.payload).pipe(
       map((resp: V) => {
         return new DaffContactSuccessSubmit();
       }),
       catchError(error => {
         return of(new DaffContactFailedSubmit(['Failed to submit']));
       })
    )
}
```

Reply:
This should be taken care of in c08df8248fc8c53e81472639126bd5feaecdf8fc

Reply:
Okay this should be addressed in 74bef0998c0734b59d30109584e192d22b6e14c5.
Let me know if those changes look good. After looking at it I do think the effect now looks more simple.